hostdb/hostdb.go

```diff
@@ -54,17 +62,41 @@ func ForEachAnnouncement(b types.Block, height types.BlockHeight, fn func(consen
 	}
 }
 
-// An Interaction represents a generic interaction with a host.
+// Interactions contains metadata about a host's interactions.
+type Interactions struct {
+	TotalScans          uint64
+	LastScan            time.Time
+	LastScanSuccess     bool
```

Comment:
hm so originally I figured this was fine, but then I encountered `IsOnline() bool` a bit further down, isn't it better to keep `LastScanSuccess` and `Online` params on the `Interactions`. Using those two fields I think you can code up all the logic to flip online if necessary and you avoid the `IsOnline` method